Log creation can be flagged as blocking which can cause test suite to fail

### Describe the bug

When initializing loggers in a class it is possible that the Log creation can acquire a lock if done concurrently with another Log being initialized. I assume this only happens with an instance Log field as class initialization is done serially. But either way we should add an exception for Logger initialization as this is not a reason to complain about blocking.

```
2025-07-07T21:57:28.1728025Z 2025-07-07T21:57:28.109524313Z non-blocking-thread-ProtobufMetadataManagerInterceptorTest-NodeA-3-3 ERROR Test failed: org.infinispan.query.remote.impl.ProtobufMetadataManagerInterceptorTest.BlockingChecker
2025-07-07T21:57:28.1734248Z java.lang.AssertionError: Blocking call! jdk.internal.misc.Unsafe#park on thread Thread[#95,non-blocking-thread-ProtobufMetadataManagerInterceptorTest-NodeA-3-3,5,ISPN-non-blocking-group]
2025-07-07T21:57:28.1737894Z 	at org.infinispan.util.CoreTestBlockHoundIntegration.lambda$applyTo$0(CoreTestBlockHoundIntegration.java:51)
2025-07-07T21:57:28.1753100Z 	at reactor.blockhound.BlockHound$Builder.lambda$install$8(BlockHound.java:488)
2025-07-07T21:57:28.1754499Z 	at reactor.blockhound.BlockHoundRuntime.checkBlocking(BlockHoundRuntime.java:89)
2025-07-07T21:57:28.1755539Z 	at java.base/jdk.internal.misc.Unsafe.park(Unsafe.java)
2025-07-07T21:57:28.1756955Z 	at java.base/java.util.concurrent.locks.LockSupport.park(LockSupport.java:221)
2025-07-07T21:57:28.1758361Z 	at java.base/java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(AbstractQueuedSynchronizer.java:754)
2025-07-07T21:57:28.1787395Z 	at java.base/java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(AbstractQueuedSynchronizer.java:990)
2025-07-07T21:57:28.1789077Z 	at java.base/java.util.concurrent.locks.ReentrantReadWriteLock$WriteLock.lock(ReentrantReadWriteLock.java:959)
2025-07-07T21:57:28.1790805Z 	at org.apache.logging.log4j.core.util.internal.InternalLoggerRegistry.computeIfAbsent(InternalLoggerRegistry.java:235)
2025-07-07T21:57:28.1792490Z 	at org.apache.logging.log4j.core.LoggerContext.getLogger(LoggerContext.java:568)
2025-07-07T21:57:28.1793687Z 	at org.apache.logging.log4j.core.LoggerContext.getLogger(LoggerContext.java:541)
2025-07-07T21:57:28.1794837Z 	at org.apache.logging.log4j.core.LoggerContext.getLogger(LoggerContext.java:71)
2025-07-07T21:57:28.1796138Z 	at org.apache.logging.log4j.LogManager.getLogger(LogManager.java:624)
2025-07-07T21:57:28.1797098Z 	at org.jboss.logging.Log4j2Logger.<init>(Log4j2Logger.java:36)
2025-07-07T21:57:28.1798166Z 	at org.jboss.logging.Log4j2LoggerProvider.getLogger(Log4j2LoggerProvider.java:36)
2025-07-07T21:57:28.1799412Z 	at org.jboss.logging.Log4j2LoggerProvider.getLogger(Log4j2LoggerProvider.java:32)
2025-07-07T21:57:28.1800423Z 	at org.jboss.logging.Logger.getLogger(Logger.java:2467)
2025-07-07T21:57:28.1801318Z 	at org.jboss.logging.Logger.doGetMessageLogger(Logger.java:2644)
2025-07-07T21:57:28.1802240Z 	at org.jboss.logging.Logger.getMessageLogger(Logger.java:2536)
2025-07-07T21:57:28.1803146Z 	at org.jboss.logging.Logger.getMessageLogger(Logger.java:2521)
2025-07-07T21:57:28.1804114Z 	at org.infinispan.util.logging.LogFactory.getLog(LogFactory.java:18)
2025-07-07T21:57:28.1805352Z 	at org.infinispan.remoting.transport.impl.MultiTargetRequest.<clinit>(MultiTargetRequest.java:27)
2025-07-07T21:57:28.1807064Z 	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.invokeCommandOnAll(JGroupsTransport.java:1014)
2025-07-07T21:57:28.1808746Z 	at org.infinispan.topology.TopologyManagementHelper.executeOnClusterSync(TopologyManagementHelper.java:44)
2025-07-07T21:57:28.1810435Z 	at org.infinispan.topology.ClusterTopologyManagerImpl.fetchClusterStatus(ClusterTopologyManagerImpl.java:593)
2025-07-07T21:57:28.1812220Z 	at org.infinispan.topology.ClusterTopologyManagerImpl.recoverClusterStatus(ClusterTopologyManagerImpl.java:471)
2025-07-07T21:57:28.1813996Z 	at org.infinispan.topology.ClusterTopologyManagerImpl.lambda$handleClusterView$6(ClusterTopologyManagerImpl.java:436)
2025-07-07T21:57:28.1815626Z 	at org.infinispan.util.concurrent.ActionSequencer.safeNonBlockingCall(ActionSequencer.java:59)
2025-07-07T21:57:28.1817207Z 	at org.infinispan.util.concurrent.ActionSequencer$SequenceEntry.run(ActionSequencer.java:228)
2025-07-07T21:57:28.1818534Z 	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173)
2025-07-07T21:57:28.1819899Z 	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166)
2025-07-07T21:57:28.1821347Z 	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
2025-07-07T21:57:28.1822623Z 	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:408)
2025-07-07T21:57:28.1823822Z 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998)
2025-07-07T21:57:28.1825097Z 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
2025-07-07T21:57:28.1826465Z 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
2025-07-07T21:57:28.1827690Z 	at java.base/java.lang.Thread.run(Thread.java:1583)
```

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Infinispan version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/commons/all/src/main/java/org/infinispan/commons/internal/CommonsBlockHoundIntegration.java
+++ b/commons/all/src/main/java/org/infinispan/commons/internal/CommonsBlockHoundIntegration.java
@@ -16,6 +16,7 @@ import org.infinispan.commons.util.ProgressTracker;
 import org.infinispan.commons.util.ServiceFinder;
 import org.infinispan.commons.util.SslContextFactory;
 import org.infinispan.commons.util.concurrent.NonBlockingRejectedExecutionHandler;
+import org.jboss.logging.Logger;
 import org.kohsuke.MetaInfServices;
 
 import reactor.blockhound.BlockHound;
@@ -105,5 +106,8 @@ public class CommonsBlockHoundIntegration implements BlockHoundIntegration {
       }
 
       builder.allowBlockingCallsInside("org.apache.logging.log4j.core.Logger", "logMessage");
+
+      // Ignore blocking when initializing a Logger as it rarely occurs
+      builder.allowBlockingCallsInside(Logger.class.getName(), "getMessageLogger");
    }
 }

